hack/rollback_ci-operator.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# Script to rollback the ci-operator version and mirror the image to QCI
+
+# Function to check required tools
+check_requirements() {
+    for cmd in oc yq jq; do
+        if ! command -v $cmd &> /dev/null; then
+            echo "Error: $cmd is not installed. Please install it and try again."
+            exit 1
+        fi
+    done
+}
+
+# Function to get the tags of ci-operator from app.ci
+get_tags() {
+    echo "Fetching tags from app.ci imagestream..."
+    oc --context app.ci -n ci get imagestreams ci-operator -o yaml | yq .status.tags
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to fetch tags from app.ci."
+        exit 1
+    fi
+}
+
+# Function to validate digest
+validate_digest() {
+    if [[ ! $1 =~ ^sha256:[a-f0-9]{64}$ ]]; then
+        echo "Error: Invalid digest format. Please provide a valid sha256 digest."
+        exit 1
+    fi
+}
+
+# Function to tag the given digest to ci-operator:latest
+tag_digest() {
+    local digest=$1
+    echo "Tagging ci-operator@$digest to ci-operator:latest..."
+    oc --context app.ci -n ci tag ci-operator@$digest ci-operator:latest
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to tag the digest to ci-operator:latest."
+        exit 1
+    fi
+}
+
+# Function to mirror the ci-operator:latest image to QCI
+mirror_image() {
+    local creds_file=$(mktemp /tmp/registry-push-credentials.XXXXXX.c)
+    trap "rm -f $creds_file" EXIT  # Ensure creds file is removed
+
+    echo "Extracting registry-push credentials..."
+    oc --context app.ci extract secret/registry-push-credentials-ci-images-mirror -n ci --to=- --keys .dockerconfigjson | jq > "$creds_file"
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to extract registry-push credentials."
+        exit 1
+    fi
+
+    echo "Starting image mirroring to QCI, this may take some time..."
+    oc image mirror --keep-manifest-list --registry-config="$creds_file" --continue-on-error=true --max-per-registry=20 \
+    registry.ci.openshift.org/ci/ci-operator:latest quay.io/openshift/ci:ci_ci-operator_latest
+
+    local mirror_status=$?
+    if [ $mirror_status -eq 0 ]; then
+        echo "Image mirroring completed successfully!"
+    else
+        echo "Image mirroring failed. Please check the logs for details."
+        exit 1
+    fi
+
+    # Clean up credentials
+    echo "Cleaning up credentials..."
+    rm -f "$creds_file"  # Remove the temporary credentials file
```

Comment:
Isn't this `rm` a duplicate of the in `trap` from above?

Reply:
yeah